src/lib/npfg/npfg.hpp

```diff
@@ -422,6 +424,7 @@ class NPFG
 	matrix::Vector2f unit_path_tangent_{matrix::Vector2f{1.0f, 0.0f}}; // unit path tangent vector
 	float signed_track_error_{0.0f}; // signed track error [m]
 	matrix::Vector2f bearing_vec_{matrix::Vector2f{1.0f, 0.0f}}; // bearing unit vector
+	matrix::Vector2f position_setpoint_{matrix::Vector2f{NAN, NAN}};
```

Comment:
I would either rename this the "closest point", or at least put that the "position setpoint" corresponds to the closest point from the vehicle on the path. Just to make a clear difference between "position setpoints" getting put in as waypoints.. and what the guidance law is actually projecting/tracking on the path.

```suggestion
	matrix::Vector2f closest_point_on_path_{matrix::Vector2f{NAN, NAN}}; // instantaneous position setpoint [m]
```